h/static/scripts/group-forms/components/test/GroupModeration-test.js

```diff
@@ -39,6 +51,65 @@ describe('GroupModeration', () => {
     });
   });
 
+  describe('annotations list', () => {
+    [true, false].forEach(loading => {
+      it('shows loading indicator when loading', () => {
+        fakeUseGroupAnnotations.returns({ loading, annotations: [] });
+        const wrapper = createComponent();
+
+        assert.equal(wrapper.exists('Spinner'), loading);
+      });
+    });
+
+    [
+      { status: 'PENDING', expectedFallbackMessage: 'You are all set!' },
+      {
+        status: 'APPROVED',
+        expectedFallbackMessage: 'No annotations found for selected status',
+      },
+      {
+        status: 'DENIED',
+        expectedFallbackMessage: 'No annotations found for selected status',
+      },
+      {
+        status: 'SPAM',
+        expectedFallbackMessage: 'No annotations found for selected status',
+      },
+    ].forEach(({ status, expectedFallbackMessage }) => {
+      it('shows fallback message when no annotations exist', () => {
+        fakeUseGroupAnnotations.returns({ loading: false, annotations: [] });
+        const wrapper = createComponent();
+
+        wrapper.find('ModerationStatusSelect').props().onChange(status);
+        wrapper.update();
+
+        assert.equal(
+          wrapper.find('[data-testid="annotations-fallback-message"]').text(),
+          expectedFallbackMessage,
+        );
+      });
+    });
+
+    it('renders every annotation that was loaded', () => {
+      const annotations = [
+        { id: '1', text: 'First annotation' },
+        { id: '2', text: 'Second annotation' },
+        { id: '3', text: 'Third annotation' },
+      ];
+      fakeUseGroupAnnotations.returns({ loading: false, annotations });
+
+      const wrapper = createComponent();
+      const annotationNodes = wrapper
+        .find('AnnotationsListContent')
+        .find('div');
```

Comment:
This should use a more specific query, otherwise it may break when the DOM structure changes in future. If we have a semantic indicator in the HTML of which are the list items (eg. attribute or particular element), we could use that. Otherwise a `data-testid` could be used.

Reply:
I have changed the element used for annotations to `article`. I think that should be specific enough for now. I'll fine-tune it as part f the task to properly render annotation cards.